rust/arrow/src/compute/kernels/aggregate.rs

```diff
@@ -141,32 +146,41 @@ where
 
     match array.data().null_buffer() {
         None => {
-            let sum = data.iter().fold(T::default_value(), |accumulator, value| {
-                accumulator + *value
-            });
+            let total = data
+                .par_iter()
```

Comment:
It's interesting that this is yielding better results. I would have thought that `rayon` being introduced at thsi level, would incur enough overhead to slow the kernels down. I've previously applied parallelism at an `Array` level instead.

Reply:
This is what MLK suggests, Array level non-primitive parallelism shared pointer overhead, that's the reason. I had so much bad time before with arc overhead in my projects.

Reply:
`rayon` is pretty smart in this respect, as its execution model starts the first part of the iterator immediately. 

I am thankful that @vertexclique introduced this, as I also believe that we greatly benefit from multi-threading at this level.

What I am a bit unsure is whether there is any issue in nested spawning: we may want to fork in other places that are nested with this op. But I am confident that someone knows the answer (👀  to @vertexclique and @alamb).

Reply:
Rayon is initializing global pool with 1.5 * logical cores. Since they are all spawned but not joined threads, that will easily work from direct users of arrow. Closures that spawn on though, are executing like fork-join. Coming to the process forking point of view, while using this library as is, won't cause problems. In the nested operations case, they will be queued and spawned to any free slot, all operations are going through global producer-consumer hub inside rayon (take a gaze into bit ops code in this pr). Sometime later if a contributor comes and says that, "I want to configure tlp" then we can just expose the pool config by a method. But since it configures itself on the machine it's running we can directly skip that part.